### PR TITLE
Add strain metadata options

### PR DESCRIPTION
**Pull request by ehennestad**
Added metadata strains as options to the Strain column.
Selected strains are coupled to the Species column and vice versa.
Slightly related, refactored handleSave

### CODE REVIEW

File: src/components/MetadataTable/MetadataTable.js [hunk 1]
```diff
@@ -203,33 +205,35 @@ export function MetadataTable(props) {
...
+    
+    var target = '/writeTable'
+
     if (process.env.NODE_ENV === "development") {
       var target_url = process.env.REACT_APP_OIDC_CLIENT_REDIRECT_URL;
-      var target = `${target_url}/writeTable`
-    }
-    else {
-      var target = '/writeTable'
+      target = `${target_url}/writeTable`
```
Comment (PolarBean) [inline src/components/MetadataTable/MetadataTable.js:234]: this should use your target variable as the second half of the URL

File: src/components/MetadataTable/MetadataTable.js [hunk 2]
```diff
@@ -203,33 +205,35 @@ export function MetadataTable(props) {
...
   const postTableData = () => {
 
+
     // this will be rewritten as a web socket eventually as its nice to have two way communication
     // also it is insane to post the entire table every time a single value is changed but YOLO
     let data = { 'table': tables, 'user': props.user["http://schema.org/alternateName"], 'project': props.project };
+    
+    var target = '/writeTable'
+
     if (process.env.NODE_ENV === "development") {
```
Comment (PolarBean) [inline src/components/MetadataTable/MetadataTable.js:232]: In general we should move away from using these if NODE_ENV== methods to set variables. If a variable needs to be different on the server then it deserves its own environment variable.

File: backend/internal/update/getRequestOptions.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    const requestOptions = {headers: requestHeader};
+    return requestOptions;
+}
+
+async function getTokenFromServiceAccount() {
+
+    let endpointURL = "https://iam.ebrains.eu/auth/realms/hbp/protocol/openid-connect/token";
+    let secret = process.env.OIDC_CLIENT_SECRET;
+    
+    let body = "grant_type=client_credentials&client_id=ebrains-wizard-dev&client_secret=" + secret + "&scope=email%20profile%20team%20group";
```
Comment (PolarBean) [inline backend/internal/update/getRequestOptions.js:26]: please use a template string

File: backend/internal/update/resolveSpeciesInStrain.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    // Add empty description if not present
+    if (!strainInstance.description) {
+        strainInstance.description = "";
+    }
+    
+});
+
+
+// Write the strain instances to a file
+savePath = process.cwd() + "/src" + "/metadata";
```
Comment (PolarBean) [inline backend/internal/update/resolveSpeciesInStrain.js:27]: please use a template string

File: backend/setup.js [hunk 5]
```diff
@@ -1,7 +1,19 @@
 const downloadOpenMINDS = require('./internal/update/download_openminds.js');
+//const fetchCoreSchemaInstances = require('./internal/update/fetchCoreSchemaInstances'); // Todo?
+//const fetchControlledTerms = require('./internal/update/fetchControlledTerms'); // Todo?
+const resolveSpeciesInStrain = require('./internal/update/resolveSpeciesInStrain'); // Todo?
 
+const getRequestOptions = require('./internal/update/getRequestOptions'); // Todo?
+
+
+resolveSpeciesInStrain
```
Comment (PolarBean) [inline backend/setup.js:9]: why is this here

File: src/components/MetadataTable/MetadataTable.js [hunk 6]
```diff
@@ -110,6 +112,8 @@ export function MetadataTable(props) {
   const { present: presentDS } = DataSource;
   let count = presentDS.length;
 
+  //console.log('presentDS', presentDS);
```
Comment (PolarBean) [inline src/components/MetadataTable/MetadataTable.js:115]: remove commented line

**Review by PolarBean — CHANGES_REQUESTED** [review]
Looks good! a few small nitpicks to address and then we will merge :)

File: backend/setup.js (hunk 5 above)
Comment (ehennestad) [reply backend/setup.js:9]: The setup method is incomplete, and should note have been part of this PR. Please accept, as I don't have time to clean up the git history now.

Comment (PolarBean) [reply backend/setup.js:9]: I more mean the line resolveSpeciesStrain seems to just be there without doing anything. Cant we just delete it?

File: src/components/MetadataTable/MetadataTable.js (hunk 1 above)
Comment (ehennestad) [reply src/components/MetadataTable/MetadataTable.js]: See 61073166d6a5dd9d564ebb56195067aac87837b4

File: src/components/MetadataTable/MetadataTable.js (hunk 2 above)
Comment (ehennestad) [reply src/components/MetadataTable/MetadataTable.js:230]: Agreed. Should make a separate issue

File: backend/internal/update/getRequestOptions.js (hunk 3 above)
Comment (ehennestad) [reply backend/internal/update/getRequestOptions.js]: Done 61073166d6a5dd9d564ebb56195067aac87837b4

File: backend/internal/update/resolveSpeciesInStrain.js (hunk 4 above)
Comment (ehennestad) [reply backend/internal/update/resolveSpeciesInStrain.js]: Used path.join See: 61073166d6a5dd9d564ebb56195067aac87837b4

File: src/components/MetadataTable/MetadataTable.js (hunk 6 above)
Comment (ehennestad) [reply src/components/MetadataTable/MetadataTable.js]: done 61073166d6a5dd9d564ebb56195067aac87837b4

**Comment by PolarBean** [comment]
Great! ready for merge
